libkbfs/block_types.go

```diff
@@ -116,12 +134,20 @@ func (db *DirBlock) Set(other Block, codec kbfscodec.Codec) {
 	if err != nil {
 		panic("Unable to DirBlock.Set")
 	}
-	*db = *dbCopy
+	db.cacheMtx.Lock()
+	defer db.cacheMtx.Unlock()
+	db.IsInd = dbCopy.IsInd
+	db.UnknownFieldSetHandler = dbCopy.UnknownFieldSetHandler
+	db.cachedEncodedSize = dbCopy.cachedEncodedSize
```

Comment:
Technically this should be be `dbCopy.GetEncodedSize()` right, so the lock is taken?  (The instance isn't accessible anywhere else so in practice it doesn't matter of course.)